azure-monitor-opentelemetry-distro/azure/monitor/opentelemetry/distro/constants.py

```diff
@@ -0,0 +1,70 @@
+from os import environ
+from pathlib import Path
+from sys import platform
+
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+_LOG_PATH_LINUX = "/var/log/applicationinsights"
+_LOG_PATH_WINDOWS = "\\LogFiles\\ApplicationInsights"
+# TODO: Enabled when duplciate logging issue is solved
+# __EXPORTER_DIAGNOSTICS_ENABLED_ENV_VAR = (
+#     "AZURE_MONITOR_OPENTELEMETRY_DISTRO_ENABLE_EXPORTER_DIAGNOSTICS"
+# )
+
+
+def _get_log_path(status_log_path=False):
+    if platform.startswith("linux"):
+        return _LOG_PATH_LINUX
+    elif platform.startswith("win32"):
+        log_path = str(Path.home()) + _LOG_PATH_WINDOWS
+        if status_log_path:
+            return log_path + "\\status"
+        else:
+            return log_path
+    else:
+        return _LOG_PATH_LINUX
+
+
+def _env_var_or_default(var_name, default_val=""):
+    try:
+        return environ[var_name]
+    except KeyError:
+        return default_val
+
+
+def __get_customer_ikey():
+    try:
+        return (
+            environ["APPLICATIONINSIGHTS_CONNECTION_STRING"]
+            .split(";")[0]
+            .split("=")[1]
+        )
+    except KeyError:
+        return "unknown"
+
+
+# TODO: Enabled when duplciate logging issue is solved
+# def __is_exporter_diagnostics_enabled():
+#     return (
+#         __EXPORTER_DIAGNOSTICS_ENABLED_ENV_VAR in environ
+#         and environ[__EXPORTER_DIAGNOSTICS_ENABLED_ENV_VAR] == "True"
+#     )
+
+
+def __is_on_app_service():
+    return "WEBSITE_SITE_NAME" in environ
+
+
+_EXTENSION_VERSION = _env_var_or_default(
+    "ApplicationInsightsAgent_EXTENSION_VERSION", "disabled"
```

Comment:
Curious, what does this env var represent?

Reply:
ApplicationInsightsAgent_EXTENSION_VERSION seems to be the flag to determine whether to monitor at least for some stacks like .NET https://learn.microsoft.com/en-us/azure/azure-monitor/app/azure-web-apps-net

Node takes the env var without defaulting, so I believe it defaults to "" for them: https://github.com/microsoft/ApplicationInsights-node.js/blob/develop/Bootstrap/DiagnosticLogger.ts

I set it to disabled as default since that is the text shown in Helen's demo: https://microsoft-my.sharepoint.com/personal/jeremyvoss_microsoft_com/_layouts/15/stream.aspx?id=/personal/jeremyvoss_microsoft_com/Documents/Recordings/TelReach+DevTeam+meeting-20220810_151758-Meeting+Recording.mp4&ga=1&sw=auth
Check timestamp 4:30 and 12:50

Reply:
Will ask Hector and App Service to see if there are clearer rules

Reply:
Confirmed with app service. Disabled should be the default.